frontend/server/src/Controllers/QualityNomination.php

```diff
@@ -359,6 +284,104 @@ public static function apiCreate(\OmegaUp\Request $r) {
+     * `promotion` and the `contents` field should be a JSON blob with the
+     * following fields:
+     *
+     * * `statements`: A dictionary of languages to objects that contain a
+     *                 `markdown` field, which is the markdown-formatted
+     *                 problem statement for that language.
+     * * `source`: A URL or string clearly documenting the source or full name
+     *             of original author of the problem.
+     * * `tags`: An array of tag names that will be added to the problem upon
+     *           promotion.
+     *
+     * # Demotion
+     *
+     * A demoted problem is banned, and cannot be un-banned or added to any new
+     * problemsets. This expects the `nomination` field to be `demotion` and
+     * the `contents` field should be a JSON blob with the following fields:
+     *
+     * * `rationale`: A small text explaining the rationale for demotion.
+     * * `reason`: One of `['duplicate', 'no-problem-statement', 'offensive', 'other', 'spam']`.
+     * * `original`: If the `reason` is `duplicate`, the alias of the original
+     *               problem.
+     * # Dismissal
+     * A user that has already solved a problem can dismiss suggestions. The
+     * `contents` field is empty.
+     *
+     * @param \OmegaUp\Request $r
+     *
+     * @return array
+     * @throws \OmegaUp\Exceptions\DuplicatedEntryInDatabaseException
+     */
+    public static function apiCreate(\OmegaUp\Request $r) {
+        if (OMEGAUP_LOCKDOWN) {
+            throw new \OmegaUp\Exceptions\ForbiddenAccessException('lockdown');
+        }
+
+        // Validate request
+        $r->ensureMainUserIdentity();
+
+        \OmegaUp\Validators::validateStringNonEmpty(
+            $r['problem_alias'],
+            'problem_alias'
+        );
+        \OmegaUp\Validators::validateInEnum(
+            $r['nomination'],
+            'nomination',
+            ['suggestion', 'promotion', 'demotion', 'dismissal']
+        );
+        \OmegaUp\Validators::validateStringNonEmpty($r['contents'], 'contents');
+        /**
+         * @var null|array{tags?: mixed, before_ac?: mixed, difficulty?: mixed, quality?: mixed, statements?: mixed, source?: mixed, reason?: mixed, original?: mixed} $contents
+         */
+        $contents = json_decode($r['contents'], true /*assoc*/);
+        if (!is_array($contents)) {
+            throw new \OmegaUp\Exceptions\InvalidParameterException(
+                'parameterInvalid',
+                'contents'
+            );
+        }
+        $problem = \OmegaUp\DAO\Problems::getByAlias($r['problem_alias']);
+        if (is_null($problem)) {
+            throw new \OmegaUp\Exceptions\NotFoundException('problemNotFound');
+        }
+
+        $nomination = \OmegaUp\Controllers\QualityNomination::createNomination(
+            $problem,
+            $r->user,
+            strval(
+                $r['nomination']
+            ),
```

Comment:
```suggestion
            strval($r['nomination']),
```

aún no logro persuadir a phpcs que si tienes llamadas a funciones anidadas, reconsidere el hecho de que algunas llamadas caben en la misma línea.